sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1243,6 +1244,18 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
         IsNotNull(e)
       case SqlBaseParser.NULL =>
         IsNull(e)
+      case SqlBaseParser.TRUE => ctx.NOT match {
+        case null => IsTrue(e)
```

Comment:
isn't it simply `EqualNullSafe(e, Literal(true))`? Why do we create a bunch of new expressions?

Reply:
@cloud-fan It looks more reasonable. Do I need to adjust? cc @dongjoon-hyun @maropu

Reply:
`EqualNullSafe` requires the left type to be the same as right type, so `EqualNullSafe(e, Literal(true))` requires `e` to be boolean.

`IsNull` is a problem so I think that change makes sense.

Reply:
Ah, I missed that approach and the @cloud-fan one sounds more reasonable. If @dongjoon-hyun is not against the idea, can you make a PR for that?

Reply:
@cloud-fan @maropu Let's wait @dongjoon-hyun , then I will make a PR.

Reply:
Sorry for being late. +1 for that approach.

Reply:
Thanks for the check, @dongjoon-hyun ! Plz go ahead, @beliefer

Reply:
OK.